### ISSUE TITLE
'Float' cannot be used as a JSX component.

### ISSUE THREAD

**Issue by iconictb** [ycs77/headlessui-float#2]
I'm getting the following error on the Float element:
`'Float' cannot be used as a JSX component.  Its return type 'Element | undefined' is not a valid JSX element.`
```
return (
<Popover>
    <Float //
        placement="top"
        offset={15}
        shift={6}
        flip
        arrow
        portal
        enter="transition duration-200 ease-out"
        enterFrom="opacity-0 -translate-y-1"
        enterTo="opacity-100 translate-y-0"
        leave="transition duration-150 ease-in"
        leaveFrom="opacity-100 translate-y-0"
        leaveTo="opacity-0 -translate-y-1"
    >
        <Popover.Button>{props.children}</Popover.Button>

        <Popover.Panel //
            className={classNames(
                //
                props.overlayClasses,
                'absolute z-50 bg-coolGray-600 rounded-md py-1 px-2  '
            )}
        >
            {props.content}
        </Popover.Panel>
    </Float>
</Popover>
)
```

**Comment by ycs77** [comment]
Is there a repo with a minimal reproducible error?

**Comment by ycs77** [comment]
This issue has not been updated for a long time, I will close it first.
If there are any problems, you can open a new issue.

**Comment by MichaelCereda** [comment]
@ycs77 I'd like to re-open the issue.


```
$ cat node_modules/headlessui-float-react/dist/float.d.ts
....
declare function FloatRoot(props: {
    show?: boolean;
    placement?: Placement;
    strategy?: Strategy;
    offset?: OffsetOptions;
    shift?: boolean | number | (ShiftOptions & DetectOverflowOptions);
    flip?: boolean | (FlipOptions & DetectOverflowOptions);
    arrow?: boolean | number;
    autoPlacement?: boolean | (AutoPlacementOptions & DetectOverflowOptions);
    hide?: boolean | (HideOptions & DetectOverflowOptions);
    autoUpdate?: boolean | AutoUpdateOptions;
    zIndex?: number;
    enter?: string;
    enterFrom?: string;
    enterTo?: string;
    leave?: string;
    leaveFrom?: string;
    leaveTo?: string;
    originClass?: string | OriginClassResolver;
    tailwindcssOriginClass?: boolean;
    portal?: boolean | string;
    transform?: boolean;
    middleware?: Middleware[] | ((refs: {
        referenceEl: MutableRefObject<Element | VirtualElement | null>;
        floatingEl: MutableRefObject<HTMLElement | null>;
    }) => Middleware[]);
    children: ReactElement[];
    onShow?: () => void;
    onHide?: () => void;
    onUpdate?: () => void;
}): JSX.Element | undefined;
...
```

The last line of the generated definition makes headless-ui Float return a JSX.Element or undefined, this triggers typescript.

**Comment by ycs77** [comment]
Release on next version

**Comment by ycs77** [comment]
@iconictb @MichaelCereda Thanks report the bug!

Now is fixed in [cca08a7](https://github.com/ycs77/headlessui-float/commit/cca08a717d1c7e924c2789b84509704e953c3f13), available in the next release.

**Comment by ycs77** [comment]
Released v0.5.0, if has any problem can open new issue.
